### PR TITLE
Rename WLAN keyword args.

### PR DESCRIPTION
**Pull request by iabdalkader**
* Rename WLAN keyword args to connect() and config() to be more consistent across ports and wlan drivers, see related RFC and discussion in #8083.
* This change is backwards compatible and will support obsolete keyword args, except for `essid` which is now deprecated in favor of `ssid`.
* Changed kwargs: `essid` changed to `ssid`, `auth` or `authmode` changed to `security` and `password` changed to `key`.

### CODE REVIEW

File: docs/library/network.WLAN.rst [hunk 1]
```diff
@@ -125,10 +125,10 @@ Methods
    Parameter      Description
    =============  ===========
    mac            MAC address (bytes)
-   essid          WiFi access point name (string)
+   ssid          WiFi access point name (string)
```
Comment (dpgeorge) [inline docs/library/network.WLAN.rst:128]: please line up the second column

Comment (iabdalkader) [reply docs/library/network.WLAN.rst]: Fixed.

**Comment by dpgeorge** [comment]
I think ninaw10 also needs some minor changes, for essid.

File: docs/library/network.WLAN.rst [hunk 2]
```diff
@@ -125,10 +125,10 @@ Methods
...
    =============  ===========
    mac            MAC address (bytes)
-   essid          WiFi access point name (string)
+   ssid           WiFi access point name (string)
    channel        WiFi channel (integer)
-   hidden         Whether ESSID is hidden (boolean)
-   authmode       Authentication mode supported (enumeration, see module constants)
+   hidden         Whether SSID is hidden (boolean)
+   security       Security protocol supported (enumeration, see module constants)
    password       Access password (string)
```
Comment (dpgeorge) [inline docs/library/network.WLAN.rst:132]: I think this needs changing to "key"

Comment (iabdalkader) [reply docs/library/network.WLAN.rst]: Replaced all the related `password` in the docs.

Comment (iabdalkader) [reply docs/library/network.WLAN.rst]: Note ESP ports use positional `ssid` and `key` not keyword args, but I fixed them anyway in their docs/quickref/tutorials.

**Comment by dpgeorge** [comment]
Rebased and merged in c502cf73e0e0df685723a828ecac1562e3be1cf0 through 6e868d47dc5dd7a90444f1173118b7d57e1e4df7

**Pull request closed without merging**